database/src/sqlite3.cpp

```diff
@@ -0,0 +1,121 @@
+#include <database.hpp>
+
+namespace Database
+{
+    Value database_sqlite_open(std::vector<Value>& args, Ark::VM* vm [[maybe_unused]])
+    {
+        if (!types::check(args, ValueType::String))
+            types::generateError(
+                "database:sqlite:open",
+                { { types::Contract { { types::Typedef("database_filename", ValueType::String) } } } },
+                args);
+
+        sqlite3* db;
+        const char* database_name = args[0].string().c_str();
+
+        if (sqlite3_open(database_name, &db) != SQLITE_OK)
+            return Nil;
+
+        return Ark::Value(Ark::UserType(db));
+    }
+
+    Value database_sqlite_close(std::vector<Value>& args, Ark::VM* vm [[maybe_unused]])
+    {
+        if (!types::check(args, ValueType::User) || !args[0].usertype().is<sqlite3>())
+            types::generateError(
+                "database:sqlite:close",
+                { { types::Contract { { types::Typedef("database", ValueType::User) } } } },
+                args);
+
+        sqlite3* db = (sqlite3*)args[0].usertypeRef().data();
```

Comment:
You can use the `as<T>()` method of the usertype to safely cast the data to an other type.